vnext/ReactUWP/Views/TextInputViewManager.cpp

```diff
@@ -302,10 +324,27 @@ void TextInputShadowNode::updateProperties(const folly::dynamic&& props)
     }
     else if (pair.first == "text")
     {
-      if (pair.second.isString())
-        textBox.Text(asHstring(pair.second));
-      else if (pair.second.isNull())
-        textBox.ClearValue(winrt::TextBox::TextProperty());
+      if (m_mostRecentEventCount == m_nativeEventCount)
+      {     
+        if (pair.second.isString())
+        {
+          auto oldValue = textBox.Text();
+          auto newValue = asHstring(pair.second);
+          if (oldValue != newValue)
+          {
+            textBox.Text(newValue);
+          }
+        }
+        else if (pair.second.isNull())
+          textBox.ClearValue(winrt::TextBox::TextProperty());
+      }
+    }
+    else if (pair.first == "mostRecentEventCount")
```

Comment:
>lse if (pair.first == "mostRecentEventCount") [](start = 5, length = 45)

I don't see any code on the javascript side of things sending updates of the mostRecentEventCount property to native.  Is the intent for consumers of TextInput to do this in their app code?  If so, this seems problematic...

Consider this scenario:
App creates a TextInput
Someone types abc into TextInput => m_nativeEventCount will increment several times
App doesn't register for any events and does nothing in response to text changed => m_mostRecentEventCount stays at 0
Sometime later the app code tries to set the text property on the TextInput to something new => this update will be ignored.

It seems like the framework should be updating m_mostRecentEventCount, but I'm not seeing how that happens.  Am I missing something?

Reply:
Here is the magic in TextInput.uwp.js, and customer doesn't need to take care of mostRecentEventCount.
It's already in our code, so you can see the code in this PR.

```
  _onChange: function (event: ChangeEvent) {
    // Make sure to fire the mostRecentEventCount first so it is already set on
    // native when the text value is set.
    if (this._inputRef && this._inputRef.setNativeProps) {
      this._inputRef.setNativeProps({
        mostRecentEventCount: event.nativeEvent.eventCount,
      });
    }
```

Reply:
Gotcha thanks - I didn't realize this code was already there.

---
In reply to: [286756728](https://github.com/microsoft/react-native-windows/pull/2521#discussion_r286756728) [](ancestors = 286756728)